timelock-impl/src/main/java/com/palantir/atlasdb/timelock/auth/impl/AuthRequirement.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.auth.impl;
+
+public enum AuthRequirement {
+    /**
+     * All namespaces are locked, only privileged clients have access.
+     */
+    ALWAYS_REQUIRE,
```

Comment:
this will only ever be true in timelock 1.x, at this point `AuthRequirement` would no longer be needed, and you'd just delete no?

`NEVER_REQUIRE` is the current state of the world, and I'm not sure how `PRIVILEGE_BASED` is disjoint from `NEVER_REQUIRE`.

if this is the break glass, it's a rather large break glass

Reply:
We wanted to keep the difference between 1.x and 0.x small, preferably just a single value rather than different implementations - as we will be supporting 0.x and 1.x for a while.
`PRIVILEGE_BASED` will be the current state of the world for 0.x; that is auth will be functional if a client publishes privileges block.
`NEVER_REQUIRE` will really be a break glass, it is actually a large break glass - but given we don't have any auth for timelock right now; it is not a regression not to use auth at all. I would like to be as safe as possible - if a client won't be able to talk to timelock for a while because of auth; it should be really easy to fix it to prevent stack wide outages.